vite.config.ts

```diff
@@ -2,14 +2,28 @@ import { defineConfig } from 'vite'
 import react from '@vitejs/plugin-react'
 
 // https://vitejs.dev/config/
-export default defineConfig({
+export default defineConfig(({ command }) => ({
   plugins: [react()],
+  root: 'src',
+  base: command === 'serve' ? '' : '/dist/',
+  build: {
+    // output dir for production build
+    outDir: '../dist',
+    emptyOutDir: true,
+    // emit manifest so PHP can find the hashed files
+    manifest: true,
+    // our entry
+    rollupOptions: {
+      input: './src/main.tsx',
```

Comment:
Since we aren't relying on `index.html` to find our entrypoints, we have to define it manually.